server/app.js

```diff
@@ -262,14 +262,26 @@ app.get('/print/:id', asyncHandler(async (req, res, next)=>{
 //Render the page
 const templateFn = require('./../client/template.js');
 app.use((req, res)=>{
+	// Create configuration object
+	const configuration = {
+		local       : false,
+		environment : config.get('node_env')
+	};
+	// Add local only items to configuration object
+	const localEnvironments = config.get('local_environments');
+	if(localEnvironments.includes(configuration.environment)){
+		configuration.local = true;
+		configuration.secret = config.get('secret');

```

Comment:
We really shouldn't be passing `secret` along to the client browser. Even in this case where it's purely for local logins, I am wary of potential abuses of this in the future.

Instead, the `localLogin` function in `account.jsx` should send a request to the server. See the Naturalcrit Repo for an example: https://github.com/naturalcrit/naturalcrit/blob/6e62368f807596dcafcbd5bcad5181bce2982455/client/naturalcrit/account.actions.js#L5-L15

Even if we don't require a password, the login should still go through the server instead of exposing key variables to the browser.

Reply:
This should now be implemented in today's changes.